platform/commonUI/edit/src/actions/CancelAction.js

```diff
@@ -82,7 +66,7 @@ define(
         CancelAction.appliesTo = function (context) {
             var domainObject = (context || {}).domainObject;
             return domainObject !== undefined &&
-                domainObject.hasCapability("editor");
+                domainObject.getCapability("status").get("editing");
```

Comment:
...looking down a little more, I think the _implementation_ of `isEditingContextRoot()` could just be a check for the `editing` status, for simplicity. Other code would still benefit from only needing to look at one capability instead of two.